spec/models/day_spec.rb

```diff
@@ -47,9 +48,52 @@
       end
 
       it 'returns students which did not attend the day' do
-        expect(subject.students_without_attendance).to eq(students(:patrik, :kaspar))
+        expect(day.students_without_attendance).to eq(students(:patrik, :kaspar))
+      end
+    end
+  end
+
+  describe '#next' do
+    context 'a day that has a next day' do
+      subject(:day) do
+        days(:day2)
+      end
+
+      it 'returns the chronologically next day' do
+        expect(day.next).to eq(days(:day3))
+      end
+    end
+
+    context 'the last day' do
+      subject(:day) do
+        days(:day3)
+      end
+
+      it 'returns nil' do
+        expect(day.next).to eq(nil)
+      end
+    end
+  end
+
+  describe '#previous' do
+    context 'a day that has previous day' do
+      subject(:day) do
+        days(:day2) # find out what days does
```

Comment:
`days` is a method defined through the line `fixtures :days, :students, :attendances` at the beginning of the file. It lets you access fixtures with a specific name (here it's a days fixture with the name "day2"). Fixtures are just records stored in the test database (when you start the tests, they're loaded automatically from the `spec/fixtures/` directory).

Reply:
Clear